### PR TITLE
Add refactor changes minus Sleep-repo, which is not working as expected

### PR DESCRIPTION
**Pull request by spaceplesiosaur**
This adds changes I made to `Activity.js`, `Activity-test.js`, `Sleep.js`, and `Sleep-test.js`, as well as `scripts.js`.

I have refactored these classes to combine matching methods into one and dry up code.  All of the tests were still working when I checked, and I changed `scripts.js` to take in the new code.

If you can check to see that the tests still pass and the page still loads, that would be great.  If everything is working, I think we can close out https://github.com/spaceplesiosaur/refactor-tractor/issues/8 and https://github.com/spaceplesiosaur/refactor-tractor/issues/17.

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -134,35 +134,35 @@ $(document).ready(function () {
   });
 
   //Sleep
-  $('.hours-slept-day').text(`${sleep.returnSleepHours(date)} hours | ${sleep.returnSleepQuality(date)} quality`);
+  $('.hours-slept-day').text(`${sleep.returnsleepData(date, 'hoursSlept')} hours | ${sleep.returnsleepData(date, 'sleepQuality')} quality`);
```
Comment (ckoga) [inline src/scripts.js]: I'm not very familiar with the bitwise-or operator (except what I just <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Bitwise_Operators">googled</a>).  I'm not saying we should change it but I definitely can't explain what is happening here.

**Comment by ckoga** [comment]
Great job @spaceplesiosaur  it was really easy to follow the changes with good naming conventions like relevantData, unfortunately I still can't follow it to see where those numbers are changing.  When I take a break from scss Ill pull this branch and see if I can figure anything out.  I put a comment in the files changed about an operator I'm not familiar with we don't need to change it but I will need to do some more research on what is actually happening there (scripts.js line 137 ).

**Comment by spaceplesiosaur** [comment]
Thanks!  Yeah the line is just part of the string and something that the previous students did.  I definitely know nothing about bitwise operators!

The numbers wouldn't be changing on this branch - I put that in a separate branch.  If this one looks good you can probably merge it and then take a look at the SleepRepo PR.  There's definitely an issue there.

File: src/Sleep.js [hunk 2]
```diff
@@ -17,42 +17,24 @@ class Sleep {
...
-  returnAvgSleepHours() {
+  returnAvgSleepData(relevantProperty) {
+      let specificUser = this.findUser();
+      return Number((specificUser.reduce((totalHours, day) => {
+        totalHours += day[relevantProperty];
+        return totalHours;
+      }, 0) / specificUser.length).toFixed(2));
+    }
+
+  returnsleepData(date, relevantProperty) {
```
Comment (MattTuring) [inline src/Sleep.js]: change the casing here

**Comment by MattTuring** [comment]
I think its ready to be merged after the very small change to line 26 in the sleep js

File: src/Sleep.js (hunk 2 above)
Comment (spaceplesiosaur) [reply src/Sleep.js]: @MattTuring By casing do you mean returnsleepData should be returnSleepData?

**Comment by spaceplesiosaur** [comment]
@MattTuring made the change!

**Pull request merged**